File: support_tools/swigwin-3.0.12/Examples/test-suite/go/threads_exception_runme.go
package main

import "strings"
import "./threads_exception"

func main() {
	t := threads_exception.NewTest()

	error := true
	func() {
		defer func() {
			error = recover() == nil
		}()
		t.Unknown()
	}()
	if error {
		panic(0)
	}

	error = true
	func() {
		defer func() {
			error = strings.Index(recover().(string), "int exception") == -1
		}()
		t.Simple()
	}()
	if error {
		panic(0)
	}

	error = true
	func() {
		defer func() {
			error = recover().(string) != "I died."
		}()
		t.Message()
	}()
	if error {
		panic(0)
	}

	error = true
	func() {
		defer func() {
			e := recover().(string)
			error = strings.Index(e, "Exc exception") == -1
		}()
		t.Hosed()
	}()
	if error {
		panic(0)
	}

	for i := 1; i < 4; i++ {
		error = true
		func() {
			defer func() {
				error = recover() == nil
			}()
			t.Multi(i)
		}()
		if error {
			panic(0)
		}
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/php/threads_exception_runme.php
<?php

require "tests.php";
require "threads_exception.php";

// Check functions
check::functions(array(test_simple,test_message,test_hosed,test_unknown,test_multi,is_python_builtin));
// Check classes.
check::classes(array(Exc,Test,threads_exception));
// Chek globals.
check::globals(array(exc_code,exc_msg));

$t = new Test();
try {
    $t->unknown();
} catch (Exception $e) {
    check::equal($e->getMessage(), 'C++ A * exception thrown', '');
}

try {
    $t->simple();
} catch (Exception $e) {
    check::equal($e->getCode(), 37, '');
}

try {
    $t->message();
} catch (Exception $e) {
    check::equal($e->getMessage(), 'I died.', '');
}

try {
    $t->hosed();
} catch (Exception $e) {
    check::equal($e->getMessage(), 'C++ Exc exception thrown', '');
}

foreach (Array(1,2,3,4) as $i) {
    try {
	$t->multi($i);
    } catch (Exception $e) {
    }
}
